Source/WebCore/platform/audio/cocoa/AudioSampleBufferConverter.mm

```diff
@@ -172,11 +172,21 @@
         m_destinationFormat.mChannelsPerFrame = m_sourceFormat.mChannelsPerFrame;
         m_destinationFormat.mSampleRate = m_sourceFormat.mSampleRate;
     }
-    if  (outputFormatID == kAudioFormatOpus)
-        m_destinationFormat.mSampleRate = 48000;
 
-    UInt32 size = sizeof(m_destinationFormat);
-    if (auto error = PAL::AudioFormatGetProperty(kAudioFormatProperty_FormatInfo, 0, NULL, &size, &m_destinationFormat)) {
+    if (auto error = [&](auto& destinationFormat) {
+        auto originalDestinationFormat = destinationFormat;
+        UInt32 size = sizeof(destinationFormat);
+        auto result = PAL::AudioFormatGetProperty(kAudioFormatProperty_FormatInfo, 0, NULL, &size, &destinationFormat);
+        if (result == kAudioCodecUnsupportedFormatError) {
+            destinationFormat.mSampleRate = 48000;
+            result = PAL::AudioFormatGetProperty(kAudioFormatProperty_FormatInfo, 0, NULL, &size, &destinationFormat);
+            if (originalDestinationFormat.mFramesPerPacket) {
+                // Adjust mFramesPerPacket to match new sampling rate
+                destinationFormat.mFramesPerPacket = originalDestinationFormat.mFramesPerPacket / originalDestinationFormat.mSampleRate * destinationFormat.mSampleRate;
```

Comment:
Why changing destinationFormat if result is false?

Reply:
it's a thrown away value if error, I thought it didn't matter.